ergm.py

```diff
@@ -481,6 +482,11 @@ def fit(self, observed_network,
             else:
                 raise ValueError(f"Convergence criterion {convergence_criterion} not defined")
 
+        del networks_for_sample
```

Comment:
You cleanup after optimization is completed, it's worth a try to indent this such that we delete at each step all the arrays that were used and generated, and create new ones.